### PR TITLE
Test enumeration support xcscheme variables

### LINKED ISSUE #11: Enable debug mode for test dumper
with verbose output , the ability to set output temp path and the abillity to set timeout

### PR DESCRIPTION
**Pull request by danielribeiro**
Also adding debugging support (fixes #11), and more robust file handling on the dylib, managing situations where the simulator launches the xctest application more than once

### CODE REVIEW

**Comment by danielribeiro** [comment]
cc @justinseanmartin  @mikelikespie

File: lib/xcknife/xcscheme_analyzer.rb [hunk 1]
```diff
@@ -7,6 +7,10 @@ module XcschemeAnalyzer
     def extract_environment_variables(xscheme_data)
       ret = {}
       xml_root = REXML::Document.new(xscheme_data).root
+      test_action = xml_root.elements["//TestAction"]
+      if test_action && test_action.attributes['shouldUseLaunchSchemeArgsEnv'] == "NO"
+        return ret
```
Comment (justinseanmartin) [inline lib/xcknife/xcscheme_analyzer.rb]: I don't think this early return is correct. I think this should be:

```
    action = xml_root.elements["//TestAction"]
    if test_action.attributes['shouldUseLaunchSchemeArgsEnv'] == "YES"
        action = xml_root.elements["//LaunchAction"]
    end
```

Comment (danielribeiro) [inline lib/xcknife/xcscheme_analyzer.rb]: they can also `have shouldUseLaunchSchemeArgsEnv == "NO"`, right?

**Comment by justinseanmartin** [comment]
Thanks, LGTM!

**Pull request merged**